src/components/Tool/ToolBanner.tsx

```diff
@@ -11,17 +12,55 @@ export function ToolBanner() {
 
   const { data } = trpc.tool.getAll.useQuery(undefined, { enabled: !!toolIds?.length });
   const selected = data?.find((x) => x.id === selectedId);
+  const theme = useMantineTheme();
 
   if (!data || !selected) return null;
 
+  const hasHeader = !!selected.metadata?.header;
+
   return (
-    <div className="-mt-4 mb-3 bg-gray-1 px-3 py-6 dark:bg-dark-9">
+    <div
+      className="relative -mt-4 mb-4 overflow-hidden bg-gray-1 px-3 py-6 dark:bg-dark-9"
+      style={
+        hasHeader
+          ? {
+              color: theme.white,
+            }
+          : undefined
+      }
```

Comment:
If we want to move out from mantine, we can probably do a css variable for this stored in the `globals.css` file
```suggestion
        hasHeader
          ? { color: 'var(--color-white)' }
          : undefined
      }
```

Then in `globals.css`
```css
:root {
  --color-white: '#fefefe',
}
```